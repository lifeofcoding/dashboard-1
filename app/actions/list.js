/* eslint-disable object-shorthand */
// @flow
import type { GetItemState, Dispatch } from '../reducers/types';

export const ADD_ITEM = 'ADD_ITEM';
export const REMOVE_ITEM = 'REMOVE_ITEM';

export const add = (payload) => {
    return (dispatch: Dispatch) => {

        if(payload.length < 1) {
            return;
        }
        dispatch({
            type: ADD_ITEM,
            payload: payload
        })
    }
}

export const remove = (payload) => {
    return (dispatch: Dispatch, getItemState: GetItemState) => {
    
        const { items }  = getItemState();
        const filteredList = items.filter( (item, i) => {
            if(i !== payload) {
                return item;
            }
        })

        dispatch({
            type: REMOVE_ITEM,
            payload: filteredList
        })
    }
}
