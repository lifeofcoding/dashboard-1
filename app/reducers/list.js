// @flow
import type { Action } from './types';
import { ADD_ITEM, REMOVE_ITEM } from "../actions/list";

// IState: {
//   items: Array<string>
// }

// initState = {
//   items: Array<string> = ["test", "test"]
// }

export default function counter(state: Array<string> = ["test", "test"], action: Action) {
  switch (action.type) {
    case ADD_ITEM: 
      return [...state, action.payload];
    case REMOVE_ITEM:
      return action.payload;
    default:
      return state;
  }
}
