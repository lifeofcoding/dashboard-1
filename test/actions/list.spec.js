import { spy } from 'sinon';
import * as actions from '../../app/actions/list';

describe('actions', () => {
  const payload = 'i love tam';
  it('should add a new item to list', () => {
    expect(actions.add(payload)).toMatchSnapshot();
  });

  it('should create remove item action', () => {
    expect(actions.remove()).toMatchSnapshot();
  })

  it('should remove item from list', () => {
    const index = 1;
    const fn = actions.remove(index);
    expect(fn).toBeInstanceOf(Function);
    const dispatch = spy();
    const getState = () => {
      return {items: ['test', 'test2']}
    };
    const filteredList = ['test']
    fn(dispatch, getState);
    // dispatch.alwaysReturned()
    expect(dispatch.calledWith({ type: actions.REMOVE_ITEM, payload: filteredList})).toBe(true);
  });

  

  // it('should incrementIfOdd should create increment action', () => {
  //   const fn = actions.incrementIfOdd();
  //   expect(fn).toBeInstanceOf(Function);
  //   const dispatch = spy();
  //   const getState = () => ({ counter: 1 });
  //   fn(dispatch, getState);
  //   expect(dispatch.calledWith({ type: actions.INCREMENT_COUNTER })).toBe(true);
  // });

  // it('should incrementIfOdd shouldnt create increment action if counter is even', () => {
  //   const fn = actions.incrementIfOdd();
  //   const dispatch = spy();
  //   const getState = () => ({ counter: 2 });
  //   fn(dispatch, getState);
  //   expect(dispatch.called).toBe(false);
  // });

  // There's no nice way to test this at the moment...
  // it('should incrementAsync', done => {
  //   const fn = actions.incrementAsync(1);
  //   expect(fn).toBeInstanceOf(Function);
  //   const dispatch = spy();
  //   fn(dispatch);
  //   setTimeout(() => {
  //     expect(dispatch.calledWith({ type: actions.INCREMENT_COUNTER })).toBe(
  //       true
  //     );
  //     done();
  //   }, 5);
  // });
});
