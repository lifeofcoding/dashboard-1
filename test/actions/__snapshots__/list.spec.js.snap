// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should add a new item to list 1`] = `[Function]`;

exports[`actions should create remove item action 1`] = `[Function]`;
